perllib/FixMyStreet/Cobrand/Peterborough.pm

```diff
@@ -131,81 +139,94 @@ around 'open311_config' => sub {
+        # Leased land - count as 'private' (i.e. not dealt with in Bartec)
+        $features = $self->_fetch_features(
+            {   type   => 'arcgis',
+                url    => 'https://peterborough.assets/3/query?',
                 buffer => 1,
             },
-            $x,
-            $y,
+            $x, $y,
         );
 
-        # if not then check if it's land leased out or on a road.
-        unless ( $features && scalar @$features ) {
-            my $leased_features = $self->_fetch_features(
-                {
-                    type => 'arcgis',
-                    url => 'https://peterborough.assets/3/query?',
-                    buffer => 1,
-                },
-                $x,
-                $y,
-            );
-
-            # some PCC land is leased out and not dealt with in bartec
-            $features = [] if $leased_features && scalar @$leased_features;
-
-            # if it's not council, or leased out land check if it's on an
-            # adopted road
-            unless ( $leased_features && scalar @$leased_features ) {
-                my $road_features = $self->_fetch_features(
-                    {
-                        buffer => 1, # metres
-                        type => 'arcgis',
-                        url => 'https://peterborough.assets/7/query?',
-                    },
-                    $x,
-                    $y,
-                );
-
-                $features = $road_features if $road_features && scalar @$road_features;
-            }
-        }
+        $land_type = 'private' if $features && @$features;
+    }
 
-        # is on land that is handled by bartec so send
-        if ( $features && scalar @$features ) {
-            return $self->SUPER::get_body_sender($body, $problem);
-        }
+    unless ($land_type) {
+        # Adopted road - count as 'public'
+        $features = $self->_fetch_features(
+            {   type   => 'arcgis',
+                url    => 'https://peterborough.assets/7/query?',
+                buffer => 1,
+            },
+            $x, $y,
+        );
 
-        # neither of those so just send email for records
-        my $emails = $self->feature('open311_email');
-        if ( $emails->{flytipping} ) {
-            $problem->set_extra_metadata('flytipping_email' => $emails->{flytipping});
+        $land_type = 'public' if $features && @$features;
+    }
 
-            # P'bro do not want to be notified of smaller incident sizes. They
-            # also do not want email for reports raised by staff.
-            return { method => 'Blackhole' }
-                if _is_small_flytipping_incident($problem)
-                || _is_raised_by_staff($problem);
+    $land_type = 'private' unless $land_type;
 
-            my $contact = $self->SUPER::get_body_sender($body, $problem)->{contact};
-            return { method => 'Email', contact => $contact};
-        }
+    $problem->set_extra_metadata( land_type => $land_type );
```

Comment:
This doesn't update the record in the DB because you're not calling `$problem->update`. This means the next time the page is loaded it'll have to do the lookups again.

I'm not certain that calling `$problem->update` is the correct/safe thing to do, however, as there's a chance the `$problem` we've been passed has unpersisted changes made elsewhere. (By that I mean if you look at all the ways this function can be called it might be as a result of a problem being loaded, having some changes applied and displayed to the user e.g. when editing a report in the admin).

I guess you could check `get_dirty_columns` and only update if there aren't any, or perhaps fetching the same row from the DB into a different variable and updating *that* as well might work, e.g.:

```perl
my $problem2 = $self->{c}->model('DB::Problem')->find({ id => $id });
$problem2->set_extra_metadata( land_type => $land_type );
$problem2->update;
```